### PR TITLE
[SYCLomatic] Fix lit case c_source_file_test/main.cu and redefine_sycl_type_a.c failed issue

### PR DESCRIPTION
**Pull request by intwanghao**

### CODE REVIEW

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 1]
```diff
@@ -158,7 +158,7 @@ void rewriteFileName(clang::tooling::UnifiedPath &FileName) {
 
 void rewriteFileName(clang::tooling::UnifiedPath &FileName,
                      const clang::tooling::UnifiedPath &FullPathName) {
-  std::string FilePath;
+  std::string FilePath = FileName.getPath().str();
```
Comment (tomflinda) [inline clang/lib/DPCT/SaveNewFiles.cpp:161]:
```suggestion
  std::string FilePath;
```
As FilePath is used to output by rewriteFileName in Line162.

Comment (tomflinda) [reply clang/lib/DPCT/SaveNewFiles.cpp:161]: As the new add code shown in Line179, FileName is use to get fileinfo with DpctGlobalInfo::getInstance().insertFile, so it cannot be empty string.

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**